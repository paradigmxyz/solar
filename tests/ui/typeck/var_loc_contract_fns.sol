struct S {
    uint x;
}

enum E {
    A
}

contract C {
    function f0(
        uint memory a2, //~ ERROR: data location can only be specified for array, struct or mapping types
        uint[] memory b2,
        S memory c2,
        S[] memory d2,
        E memory e2, //~ ERROR: data location can only be specified for array, struct or mapping types
        E[] memory f22
    ) private {}
    function f1(
        uint memory a2, //~ ERROR: data location can only be specified for array, struct or mapping types
        uint[] memory b2,
        S memory c2,
        S[] memory d2,
        E memory e2, //~ ERROR: data location can only be specified for array, struct or mapping types
        E[] memory f22
    ) internal {}
    function f2(
        uint memory a2, //~ ERROR: data location can only be specified for array, struct or mapping types
        uint[] memory b2,
        S memory c2,
        S[] memory d2,
        E memory e2, //~ ERROR: data location can only be specified for array, struct or mapping types
        E[] memory f22
    ) public {}
    function f3(
        uint memory a2, //~ ERROR: data location can only be specified for array, struct or mapping types
        uint[] memory b2,
        S memory c2,
        S[] memory d2,
        E memory e2, //~ ERROR: data location can only be specified for array, struct or mapping types
        E[] memory f22
    ) external {}

    function r0() private returns (
        uint memory a2, //~ ERROR: data location can only be specified for array, struct or mapping types
        uint[] memory b2,
        S memory c2,
        S[] memory d2,
        E memory e2, //~ ERROR: data location can only be specified for array, struct or mapping types
        E[] memory f22
    ) {}
    function r1() internal returns (
        uint memory a2, //~ ERROR: data location can only be specified for array, struct or mapping types
        uint[] memory b2,
        S memory c2,
        S[] memory d2,
        E memory e2, //~ ERROR: data location can only be specified for array, struct or mapping types
        E[] memory f22
    ) {}
    function r2() public returns (
        uint memory a2, //~ ERROR: data location can only be specified for array, struct or mapping types
        uint[] memory b2,
        S memory c2,
        S[] memory d2,
        E memory e2, //~ ERROR: data location can only be specified for array, struct or mapping types
        E[] memory f22
    ) {}
    function r3() external returns (
        uint memory a2, //~ ERROR: data location can only be specified for array, struct or mapping types
        uint[] memory b2,
        S memory c2,
        S[] memory d2,
        E memory e2, //~ ERROR: data location can only be specified for array, struct or mapping types
        E[] memory f22
    ) {}
}
