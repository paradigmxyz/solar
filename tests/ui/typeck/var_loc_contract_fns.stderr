error: data location can only be specified for array, struct or mapping types
  --> ROOT/tests/ui/typeck/var_loc_contract_fns.sol:LL:CC
   |
LL |         uint memory a2,
   |         ^^^^^^^^^^^^^^
   |

error: data location can only be specified for array, struct or mapping types
  --> ROOT/tests/ui/typeck/var_loc_contract_fns.sol:LL:CC
   |
LL |         E memory e2,
   |         ^^^^^^^^^^^
   |

error: data location can only be specified for array, struct or mapping types
  --> ROOT/tests/ui/typeck/var_loc_contract_fns.sol:LL:CC
   |
LL |         uint memory a2,
   |         ^^^^^^^^^^^^^^
   |

error: data location can only be specified for array, struct or mapping types
  --> ROOT/tests/ui/typeck/var_loc_contract_fns.sol:LL:CC
   |
LL |         E memory e2,
   |         ^^^^^^^^^^^
   |

error: data location can only be specified for array, struct or mapping types
  --> ROOT/tests/ui/typeck/var_loc_contract_fns.sol:LL:CC
   |
LL |         uint memory a2,
   |         ^^^^^^^^^^^^^^
   |

error: data location can only be specified for array, struct or mapping types
  --> ROOT/tests/ui/typeck/var_loc_contract_fns.sol:LL:CC
   |
LL |         E memory e2,
   |         ^^^^^^^^^^^
   |

error: data location can only be specified for array, struct or mapping types
  --> ROOT/tests/ui/typeck/var_loc_contract_fns.sol:LL:CC
   |
LL |         uint memory a2,
   |         ^^^^^^^^^^^^^^
   |

error: data location can only be specified for array, struct or mapping types
  --> ROOT/tests/ui/typeck/var_loc_contract_fns.sol:LL:CC
   |
LL |         E memory e2,
   |         ^^^^^^^^^^^
   |

error: data location can only be specified for array, struct or mapping types
  --> ROOT/tests/ui/typeck/var_loc_contract_fns.sol:LL:CC
   |
LL |         uint memory a2,
   |         ^^^^^^^^^^^^^^
   |

error: data location can only be specified for array, struct or mapping types
  --> ROOT/tests/ui/typeck/var_loc_contract_fns.sol:LL:CC
   |
LL |         E memory e2,
   |         ^^^^^^^^^^^
   |

error: data location can only be specified for array, struct or mapping types
  --> ROOT/tests/ui/typeck/var_loc_contract_fns.sol:LL:CC
   |
LL |         uint memory a2,
   |         ^^^^^^^^^^^^^^
   |

error: data location can only be specified for array, struct or mapping types
  --> ROOT/tests/ui/typeck/var_loc_contract_fns.sol:LL:CC
   |
LL |         E memory e2,
   |         ^^^^^^^^^^^
   |

error: data location can only be specified for array, struct or mapping types
  --> ROOT/tests/ui/typeck/var_loc_contract_fns.sol:LL:CC
   |
LL |         uint memory a2,
   |         ^^^^^^^^^^^^^^
   |

error: data location can only be specified for array, struct or mapping types
  --> ROOT/tests/ui/typeck/var_loc_contract_fns.sol:LL:CC
   |
LL |         E memory e2,
   |         ^^^^^^^^^^^
   |

error: data location can only be specified for array, struct or mapping types
  --> ROOT/tests/ui/typeck/var_loc_contract_fns.sol:LL:CC
   |
LL |         uint memory a2,
   |         ^^^^^^^^^^^^^^
   |

error: data location can only be specified for array, struct or mapping types
  --> ROOT/tests/ui/typeck/var_loc_contract_fns.sol:LL:CC
   |
LL |         E memory e2,
   |         ^^^^^^^^^^^
   |

error: aborting due to 16 previous errors

